src/libs/ReportUtils.ts

```diff
@@ -3065,10 +3065,41 @@ function getInvoicePayerName(report: OnyxEntry<Report>): string {
     return getPolicyName(report, false, allPolicies?.[`${ONYXKEYS.COLLECTION.POLICY}${invoiceReceiver?.policyID}`]);
 }
 
+/**
+ * Parse html of reportAction into text
+ */
+function parseReportActionHtmlToText(reportAction: OnyxEntry<ReportAction>, reportID: string): string {
+    const {html, text} = reportAction?.message?.[0] ?? {};
+
+    if (!html) {
+        return text ?? '';
+    }
+
+    const mentionReportRegex = /<mention-report reportID="(\d+)" *\/>/gi;
+    const matches = html.matchAll(mentionReportRegex);
+
+    const reportIdToName: Record<string, string> = {};
+    for (const match of matches) {
+        if (match[1] !== reportID) {
+            // eslint-disable-next-line @typescript-eslint/no-use-before-define
+            reportIdToName[match[1]] = getReportName(getReport(match[1])) ?? '';
+        }
+    }
```

Comment:
Instead of searching for mentioned room name id before  parsing, we could pass callback `(id) => getReportName(getReport(id))` into extras. In [here](https://github.com/Expensify/expensify-common/blob/84ac0394ab1021d3d892338c4dbb3f87c05b0bc9/lib/ExpensiMark.js#L536). 

But this will need changes in ExpensiMark.js